pagopa/io-app
# fix: [IOPID-2151] Fix session refresh

## Short description
This pull request addresses an issue where the session token was not being refreshed after two minutes of inactivity, ensuring that the identification process functions correctly under these conditions.

## List of changes proposed in this pull request
- Add action `areTwoMinElapsedFromLastActivity` and value on redux state `areAlreadyTwoMinAfterLastActivity` in order to keep track of whether or not the session should be refreshed
- Add logic on identificationModal 
- Delete old logic

## Demo 

<p>

| Identification after 2 min | Identification after 2 min DEVICE |
| - | - |
| <video src="https://github.com/user-attachments/assets/31d0a513-bb6b-42cc-89e0-fd7ad3e07a95"/> | <video src="https://github.com/user-attachments/assets/9a98debc-6ea4-4a9a-aa9d-b1008570fc10"/> |

| Identification after 11 min | Identification after 11 min DEVICE |
| - | - |
| <video src="https://github.com/user-attachments/assets/381006eb-c228-4863-ad53-1e81c5edbc75"/> | <video src="https://github.com/user-attachments/assets/ea4f0577-71f1-4a56-8f82-dc3d7f7bef33"/> |
</p>

## How to test
- Run the application with production environment, 
- log into the app, 
- Active toggle "Active session refresh" on profile section,
- put the app in background for at least 2 minutes (11 minutes are better) and when you put the app in foreground you must check whether the session refresh calls are executed.
